spec/models/order_number/definition_spec.rb

```diff
@@ -38,4 +38,84 @@
   it_behaves_like 'an entity that has goods nomenclatures' do
     let(:entity) { build(:definition, measures:) }
   end
+
+  describe 'suspension_period?' do
+    subject { order_definition.suspension_period? }
+
+    context 'with start and end dates' do
+      let :order_definition do
+        build :definition, suspension_period_start_date: Date.yesterday.xmlschema,
+                           suspension_period_end_date: Date.tomorrow.xmlschema
+      end
+
+      it { is_expected.to be true }
+    end
+
+    context 'with only start date' do
+      let :order_definition do
+        build :definition, suspension_period_start_date: Date.yesterday.xmlschema,
+                           suspension_period_end_date: nil
+      end
+
+      it { is_expected.to be false }
+    end
+
+    context 'with only end date' do
+      let :order_definition do
+        build :definition, suspension_period_start_date: nil,
+                           suspension_period_end_date: Date.tomorrow.xmlschema
+      end
+
+      it { is_expected.to be false }
+    end
+
+    context 'with no start or end dates' do
+      let :order_definition do
+        build :definition, suspension_period_start_date: nil,
+                           suspension_period_end_date: nil
+      end
+
+      it { is_expected.to be false }
+    end
+  end
+
+  describe 'blocking_period?' do
+    subject { definition.blocking_period? }
+
+    context 'with start and end dates' do
+      let :definition do
+        build :definition, blocking_period_start_date: Date.yesterday.xmlschema,
+                           blocking_period_end_date: Date.tomorrow.xmlschema
+      end
+
+      it { is_expected.to be true }
+    end
+
+    context 'with only start date' do
+      let :definition do
+        build :definition, blocking_period_start_date: Date.yesterday.xmlschema,
+                           blocking_period_end_date: nil
+      end
+
+      it { is_expected.to be false }
+    end
+
+    context 'with only end date' do
+      let :definition do
+        build :definition, blocking_period_start_date: nil,
+                           blocking_period_end_date: Date.tomorrow.xmlschema
+      end
+
+      it { is_expected.to be false }
+    end
+
+    context 'with no start or end dates' do
+      let :definition do
+        build :definition, blocking_period_start_date: nil,
+                           blocking_period_end_date: nil
+      end
+
+      it { is_expected.to be false }
+    end
+  end
```

Comment:
very readable tests 👍 
I just noticed that maybe we  could remove some of the tests,
I guess we could tests only 2 conditions, true (when both are true)
and false (when one is false)
instead of
true true
true false
false true
false false

Up to you anyway 😉

Reply:
I'd disagree with this -> my reasoning being the tests should testing combinations of input conditions not outcomes

There are 4 possible combinations of inputs
There are 2 possible outcomes

If I was writing a test for `String#blank?` - it only has `true` and `false` as outcomes but if I only test it with an empty string I'm still not proving that it also returns `true` for a string consistenting only of whitespace.

An implementation as follows would pass the test suite but actually return `false` for a whitespace string - this would mean the test suite was incomplete?

```ruby
def blank?
  nil? || length.zero?
end
```